### PR TITLE
Make the detector compatible with both smart-agent and otel agent

### PR DESCRIPTION
**Pull request by maxenced**

### CODE REVIEW

File: modules/smart-agent_redis/variables.tf [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Module specific
 
+variable "from_otel" {
+  description = "Set to true when metrics are sent by otel collector instead of smart-agent"
+  type        = bool
+  default     = false
```
Comment (Shr3ps) [inline modules/smart-agent_redis/variables.tf:6]: isn't otel now the new default? to set to true by default?
(or maybe reverse the variable meaning: "use legacy" ?)

**Review by xp-1000 — CHANGES_REQUESTED** [review]

File: modules/smart-agent_redis/variables.tf (hunk 1 above)
Comment (maxenced) [reply modules/smart-agent_redis/variables.tf:6]: given that the module is named `smart-agent_redis`, I would keep the default behavior to be smart-agent (+ it avoids a breaking change)

File: modules/smart-agent_redis/detectors-redis.tf [hunk 2]
```diff
@@ -176,15 +176,16 @@ EOF
 }
 
 resource "signalfx_detector" "memory_used_max" {
-  name = format("%s %s", local.detector_name_prefix, "Redis memory used over max memory (if configured)")
+  count = var.from_otel ? 0 : 1
```
[the remark this answers is not in the document]
Comment (maxenced) [reply modules/smart-agent_redis/detectors-redis.tf:179]: Yep, max memory is not exposed by otel collector (or i missed it).

File: modules/smart-agent_redis/variables.tf [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Module specific
 
+variable "from_otel" {
+  description = "Set to true when metrics are sent by otel collector instead of smart-agent"
```
[the remark this answers is not in the document]
Comment (maxenced) [reply modules/smart-agent_redis/variables.tf]: renamed variable and updated doc .

File: modules/smart-agent_redis/locals.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  evicted_keys_metric_name         = var.from_otel ? "redis.keys.evicted" : "counter.evicted_keys"
+  expired_keys_metric_name         = var.from_otel ? "redis.keys.expired" : "counter.expired_keys"
+  blocked_clients_metric_name      = var.from_otel ? "redis.client.blocked" : "gauge.blocked_clients"
+  connected_clients_metric_name    = var.from_otel ? "redis.client.connected" : "gauge.connected_clients"
+  rejected_connections_metric_name = var.from_otel ? "redis.connections.rejected" : "counter.rejected_connections"
+  keyspace_hits_metric_name        = var.from_otel ? "redis.keyspace.hits" : "counter.keyspace_hits"
+  keyspace_misses_metric_name      = var.from_otel ? "redis.keyspace.hits" : "counter.keyspace_hits"
+  db_keys_metric_name              = var.from_otel ? "redis.db.keys" : "gauge.db0_keys"
+
+  plugin_filter = var.from_otel ? "" : "filter('plugin', 'redis_info') and "
```
[the remark this answers is not in the document]
Comment (xp-1000) [reply modules/smart-agent_redis/locals.tf]: ok thanks

File: modules/smart-agent_redis/detectors-redis.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (xp-1000) [reply modules/smart-agent_redis/detectors-redis.tf]: ok we will need to add this in the generator in future.

you are lucky this module did not already use it :p

File: modules/smart-agent_redis/variables.tf (hunk 1 above)
Comment (xp-1000) [reply modules/smart-agent_redis/variables.tf:6]: yes please keep false

File: modules/smart-agent_redis/conf/readme.yaml [hunk 5]
```diff
@@ -21,3 +21,8 @@ notes: |
   * The "keyspace full" detector uses number of keys from database index 0,
   it will not work for other databases (1-15 by default).
   This detector is disabled by default because it makes sens only when redis is used as cache.
+  * This module supports the `from_otel` variable in case you are monitoring Redis with otel collector.
```
Comment (xp-1000) [inline modules/smart-agent_redis/conf/readme.yaml:24]: it is not `from_otel` anymore

**Review by xp-1000 — APPROVED** [review]
I runned the auto update for you to avoid another stalen PR whrereas it seems technically good enough for me

**Pull request merged**